### PR TITLE
Fix spending over time data for animated map

### LINKED ISSUE #104: Animated bucket doesn't drain
## Overview
The animated spending bucket over DC is supposed to drain. It should also agree in amount with the spending tracker at the top of the page.

### Expected Behavior
See above

### Actual Behavior
It doesn't drain, it fluctuates in value slightly, and it sometimes disappears altogether.

### PR DESCRIPTION
**Pull request by travissouthard**
## Overview

These changes fix how spending over months for each state is calculated and adds a total spending amount to get the AnimatedTotalSpendingBucket to "drain" correctly. Before, the aggregated spending was per capita by state and the total it was subtracting from was the raw total in the BIL, meaning it didn't go down by very much at all.

There was also a bug in the aggregation math that was adding too little each month and thus the coloring on the map's states didn't match the above-the-fold map at the end of the animation. 

Closes #104 

### Demo

image

### Notes

Not sure if it's because of the particular start time we are using but the bucket starts at 418B instead of 550B

image

## Testing Instructions

- In this branch delete `src/app/sr/data/monthly.spending.json` (This may be unnecessary but the fetch data script would skip if the file existed, unless there is a flag I didn't find, which, if so, do that instead)
- run `./scripts/fetch-data` (I sometimes had to do this twice because of an exit with code 1, but no output?)
- run `./scripts/server` and navigate to the homepage
- Scroll down and hit the play button on the map and ensure:
  - [x] The app plays as it normally does
  - [x] The bucket drains and matches the expected remaining amount at the end (268B)
  - [x] At the end of the animation, the animated map has the same coloring as the top map

 ## Checklist

- [ ] `fixup!` commits have been squashed
- [ ] `CHANGELOG.md` updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines
- [ ] `README.md` updated if necessary to reflect the changes
- [ ] CI passes after rebase

### CODE REVIEW

File: src/app/dataScripts/fetchSpendingOverTimeData.ts [hunk 1]
```diff
@@ -112,15 +112,17 @@ function cleanDataDump(
     const perCapitaSpendingOverMonths: MonthlySpendingOverTime = [];
     dataDump.results.reduce(
         (sum, { aggregated_amount, time_period: { fiscal_year, month } }) => {
-            const aggregatedPerCapita = sum + aggregated_amount / population;
+            const aggregatedTotal = sum + aggregated_amount;
+            const aggregatedPerCapita = (sum + aggregated_amount) / population;
```
Comment (jacobtylerwalls) [inline src/app/dataScripts/fetchSpendingOverTimeData.ts]:
```suggestion
            const aggregatedPerCapita = aggregatedTotal / population;
```

File: src/app/src/components/AnimatedArcsAndMap/AnimatedTotalSpendingBucket.tsx [hunk 2]
```diff
@@ -36,7 +36,7 @@ export default function AnimatedTotalSpendingBucket({
             const totalAwardsAtTime: number = Object.values(
                 spendingAtTimeByState
             ).reduce((sum, stateResults) => {
-                sum += stateResults && stateResults.per_capita;
+                sum += (stateResults && stateResults?.total) || 0;
```
Comment (jacobtylerwalls) [inline src/app/src/components/AnimatedArcsAndMap/AnimatedTotalSpendingBucket.tsx]:
```suggestion
                sum += stateResults?.total ?? 0;
```

**Review by jacobtylerwalls — COMMENTED** [review]
Looking good. Did you look into why the bucket starts at less than the total? There shouldn't be any infrastructure spending before the infrastructure bill was passed!

> There was also a bug in the aggregation math that was adding too little each month and thus the coloring on the map's states didn't match the above-the-fold map at the end of the animation.

Also, do you mind referring me to where you made that change ☝️ ? I thought that was already addressed in #96.

Some cosmetic feedback at your option :-)

File: src/app/dataScripts/fetchSpendingOverTimeData.ts (hunk 1 above)
Comment (travissouthard) [reply src/app/dataScripts/fetchSpendingOverTimeData.ts]: This is where I was seeing that bug before, but in retrospect, this was a misinterpretation of "It should also agree in amount with the spending tracker at the top of the page." from #104, and confusing that with the app before #96. It did gets fixed with that, so no worries.

File: src/app/src/components/AnimatedArcsAndMap/AnimatedTotalSpendingBucket.tsx (hunk 2 above)
Comment (travissouthard) [reply src/app/src/components/AnimatedArcsAndMap/AnimatedTotalSpendingBucket.tsx]: > Looking good. Did you look into why the bucket starts at less than the total? There shouldn't be any infrastructure spending before the infrastructure bill was passed!

It looks to be a data problem. The data we are seeing in the map is set to `{ "fiscal_year" : 2021, "month" : 1 }`and in the `monthly.spending.json` the earliest I see money in Alaska is month 12, 2016:
```json
{
    "per_capita": 0,
    "total": 0,
    "time_period": {
        "fiscal_year": 2016,
        "month": 11
    }
},
{
    "per_capita": 0.4374303699704051,
    "total": 320000,
    "time_period": {
        "fiscal_year": 2016,
        "month": 12
    }
},
```

Money in Idaho starts in month 4 of 2008:
```json
{
    "per_capita": 0,
    "total": 0,
    "time_period": {
        "fiscal_year": 2008,
        "month": 3
    }
},
{
    "per_capita": 53.97577155279746,
    "total": 96458212.19,
    "time_period": {
        "fiscal_year": 2008,
        "month": 4
    }
},
```

On develop it's also showing this in Idaho:
```json
{
    "per_capita": 53.97577155279746,
    "time_period": {
        "fiscal_year": 2008,
        "month": 4
    }
},
{
    "per_capita": 170.90665833083855,
    "time_period": {
        "fiscal_year": 2008,
        "month": 5
    }
},
```
And in Alaska ion develop:
```json
{
    "per_capita": 0,
    "time_period": {
        "fiscal_year": 2016,
        "month": 11
    }
},
{
    "per_capita": 0.4374303699704051,
    "time_period": {
        "fiscal_year": 2016,
        "month": 12
    }
},
```

Comment (travissouthard) [reply src/app/src/components/AnimatedArcsAndMap/AnimatedTotalSpendingBucket.tsx]: @jacobtylerwalls if we are seeing this on develop, is it worth spinning that out to its own card?

Comment (jacobtylerwalls) [reply src/app/src/components/AnimatedArcsAndMap/AnimatedTotalSpendingBucket.tsx]: Yeah, that's wise. The spending codes "Z" and "1" are supposed to exclusively deal with BIL spending, so we need an investigation card to see if we botched something or if the API is returning bad results (or if a rogue agency submitted bad contract data, hah). Would you mind opening the card?

**Review by jacobtylerwalls — APPROVED** [review]
Follow-up card makes sense for the starting level in the bucket. Nice work.

File: src/app/src/components/AnimatedArcsAndMap/AnimatedTotalSpendingBucket.tsx (hunk 2 above)
Comment (travissouthard) [reply src/app/src/components/AnimatedArcsAndMap/AnimatedTotalSpendingBucket.tsx]: Not at all. Will do so first thing.

**Pull request merged**